transaction/core/src/encrypted_fog_hint.rs

```diff
@@ -103,12 +103,31 @@ impl EncryptedFogHint {
     /// random plaintext. There are several ways we could sample that distribution
     /// but the simplest is to do exactly that. This is also future-proof if we later
     /// tweak the cryptobox implementation.
-    pub fn fake_onetime_hint<T: RngCore + CryptoRng>(rng: &mut T) -> Self {
+    /// Optionally provide the random plaintext buffer to be encrypted.
+    pub fn fake_onetime_hint<T: RngCore + CryptoRng>(
+        rng: &mut T,
+        plaintext_buf: Option<&[u8]>,
+    ) -> Self {
         // Make plaintext of the right size
-        let plaintext = GenericArray::<
-            u8,
-            Diff<EncryptedFogHintSize, <VersionedCryptoBox as CryptoBox<Ristretto>>::FooterSize>,
-        >::default();
+        let plaintext = if let Some(pt) = plaintext_buf {
+            // Truncate the plaintext to a buffer of the correct size
+            GenericArray::<
+                u8,
+                Diff<
+                    EncryptedFogHintSize,
+                    <VersionedCryptoBox as CryptoBox<Ristretto>>::FooterSize,
+                >,
+            >::clone_from_slice(pt)
+        } else {
+            GenericArray::<
+                u8,
+                Diff<
+                    EncryptedFogHintSize,
+                    <VersionedCryptoBox as CryptoBox<Ristretto>>::FooterSize,
+                >,
+            >::default()
+        };
+
```

Comment:
can we just make this a separate function?
it seems like this alternate mode of fake one-time hint will almost never be used, and so making it part of the API for the version that it used all the time seems not so good

Reply:
sgtm 👍